src/frequenz/channels/timer.py

```diff
@@ -664,7 +683,12 @@ async def ready(self) -> bool:  # noqa: DOC502
         # could be reset while we are sleeping, in which case we need to recalculate
         # the time to the next tick and try again.
         while time_to_next_tick > 0:
-            await asyncio.sleep(time_to_next_tick / 1_000_000)
+            for fin in asyncio.as_completed(
+                [asyncio.sleep(time_to_next_tick / 1_000_000), self._reset_event.wait()]
+            ):
+                await fin
+                break
```

Comment:
I was a bit confused by this `break` too (I somehow thought it was `break`ing the outer `while` instead of the `for`).

Maybe it could be a bit more clear like this?
```suggestion
            await next(asyncio.as_completed(
                [asyncio.sleep(time_to_next_tick / 1_000_000), self._reset_event.wait()]
            ))
```